From fc1cd1827d8f90e5b9d139ee96055b68ca4e0e67 Mon Sep 17 00:00:00 2001
From: Mateusz Paprocki <mattpap@gmail.com>
Date: Sat, 8 Feb 2020 22:33:45 +0100
Subject: [PATCH] Set windows selector event loop policy only if available

---
 tornado/platform/asyncio.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tornado/platform/asyncio.py b/tornado/platform/asyncio.py
index 99cd01a1..34765f15 100644
--- a/tornado/platform/asyncio.py
+++ b/tornado/platform/asyncio.py
@@ -30,7 +30,7 @@ import asyncio
 
 import sys
 
-if sys.platform == 'win32':
+if sys.platform == 'win32' and hasattr(asyncio, "WindowsSelectorEventLoopPolicy"):
     asyncio.set_event_loop_policy(asyncio.WindowsSelectorEventLoopPolicy())
 
 import typing
-- 
2.17.1

